### ISSUE TITLE
Make crypto/olm package optional: separate module, or build constraints

### ISSUE THREAD

**Issue by 08d2** [mautrix/go#71]
This was previously reported in #17, where @tulir [responded](https://github.com/mautrix/go/issues/17#issuecomment-670569926) that 

> [crypto/olm] already is optional: the crypto module isn't imported from any other module

crypto and crypto/olm are not modules themselves, they're packages within the mautrix/go module, and the rules which govern which packages get compiled are a bit subtle. There are some "tree-shaking" style optimizations at the compilation stage which can elide packages within a module that don't get directly imported, but they're more or less implementation details, so authors can't rely on them to manage optionality in this way.

In short: every package in a module is "part" of that module. Optionality has to be expressed at the module level — i.e. make crypto/olm a separate module altogether — or, if it works, as build constraints in the package — i.e. no-op implementations for unsupported architectures.

**Comment by tulir** [comment]
There's a pure Go olm alternative now, so if you use the `goolm` build tag, it won't need cgo/libolm. Other than that I'll just rely on the implementation details to optimize unused packages away.
